nhs/data/geography.py

```diff
@@ -0,0 +1,295 @@
+      and latitude) relates to the Earth's surface. We use CRS 7844 as that's what's
+      specified in the ABS 2021 datapack. As it's subject to change we'll need it in the
+      configurations.yml file for future user specification.
+    - 'EPSG:7844' is an example of a CRS in the EPSG system (European Petroleum Survey Group).
+      This is a standardized system for defining geospatial coordinates.
+    - WGS84 (EPSG:4326) is a widely used geographic CRS, representing coordinates as latitude
+      and longitude on a spherical model of the Earth.
+    """  # noqa: E501
+
+    if "LONGITUDE" in df.columns and "LATITUDE" in df.columns:
+        df = df.rename({"LONGITUDE": "x", "LATITUDE": "y"})
+
+    gdf = gpd.GeoDataFrame(
+        df.to_pandas(),
+        geometry=gpd.points_from_xy(df["x"], df["y"]),
+    )
+    projected_crs = CRS.from_string(crs)
+    return gdf.to_crs(projected_crs)
+
+
+def process_chunk(
+    map_data: gpd.GeoDataFrame,
+    chunk: gpd.GeoDataFrame,
+    area_column: str = "SA2_NAME21",
+    area_code_column: str = "SA1_CODE21",
+) -> pl.DataFrame:
+    """
+    Process a chunk of point data, spatially joining it with area polygon data.
+
+    Parameters
+    ----------
+    map_data : gpd.GeoDataFrame
+        A GeoPandas DataFrame containing spatial polygon data that represents areas (e.g., districts or regions).
+        This data includes geometry (polygons) and area-specific attributes such as 'SA2_NAME21' and 'SA1_CODE21',
+        which represent area names and codes.
+
+    chunk : gpd.GeoDataFrame
+        A GeoPandas GeoDataFrame containing point data (e.g., latitude and longitude) that needs to be joined with
+        the corresponding areas in `map_data`. This should contain a 'geometry' column representing points.
+
+    Returns
+    -------
+    pl.DataFrame
+        A Polars DataFrame with the point data from the `chunk`, joined with the appropriate area data from
+        `map_data`. The output includes the original coordinates ('x', 'y') and the corresponding area names
+        ('SA2_NAME21') and area codes ('SA1_CODE21'), renamed as 'area' and 'area_code', respectively.
+
+    Notes
+    -----
+    - The function first performs a spatial join using the 'within' predicate to match points with polygons
+      in the `map_data`. This returns the areas where each point is located.
+    - If any points cannot be matched with an area (i.e., points that do not fall within any polygon), a
+      nearest neighbor spatial join is performed to assign the closest area.
+    - The final result includes only the 'x', 'y' coordinates and the corresponding area information, and it
+      is returned as a Polars DataFrame.
+    - The function converts the intermediate GeoDataFrame to a dictionary before constructing a Polars DataFrame
+      to ensure compatibility with the Polars framework.
+    - The name is a SA2 variable as SA1s don't have names. They do of course have codes, though, which we use.
```

Comment:
Some parts are not needed as they go into too much detail on what the function is doing underneath. Try to only mention context or caveats that the end-user would care about, e.g. what are the values they should input into the function, what happens if the input lacks XTZ... 

What predicate the function uses or whether or not the function converts the data into a dictionary is not important - if it IS (e.g. when something is broken), they are better off reading the actual code instead of the doc-string. 

Should aim to keep doc-string concise as well - if you generated it with chatgpt or claude, double check (I prefer copilot since it generates line-by-line so I can check what it's using as it's generating docstrings)